### DISCUSSION TITLE
Price of Power

### DISCUSSION THREAD

**Discussion by zikronix** [jasonacox/Powerwall-Dashboard#58] [Q&A]
Hi,

I just want to verify I have this correct. The average price of my power is .10 does that mean on the savings panel I would change all values from .19 to .10 

eg:

SELECT s - (tp + tg) FROM (SELECT sum("solar") * 0.10 AS s,  sum("to_pw")  * 0.10 AS tp, sum("to_grid") * 0.10 AS tg FROM kwh.http WHERE  $timeFilter tz('America/Phoenix'))

**Comment by jasonacox** [comment]
Hi @zikronix  - that is correct!  Change the 0.19 values to 0.10.   

If anyone has a good idea on how to set those outside the query (or as an external value), please let me know if submit a PR.  🙂

**Reply by hulkster** [reply]
Ditto that it would be nice if settable in just one place. Even cooler would be if it could somehow be time-based for TOU analysis. 

I realize that is a giant can-o-worms!    ;-)

**Reply by jgleigh** [reply]
The buy/sell prices are now variables that can be setup for the whole dashboard.

**Reply by ThePnuts** [reply]
You can create a global variable that's set at the top of the dashboard. When I remade everything in Flux, I did this. Looks like this
![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/126296751/096902f6-3134-4083-a8bb-e92e3105c000)
I dunno if I can do this since im on a custom version, but her is what I have:

Variable for the dashboard
![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/126296751/849e7bc5-cdd6-4a69-ac83-18d12cde8bcb)

My code, you can see the variable used, just in flux instead. ${kWhCost} I dont think it would be to hard, just create the variable, then edit all .19 values to ${kWhCost}. Then save and export or whatever to add it how the current dashboard is.
```
  from(bucket: "tesla_energy")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "kwh")
  |> filter(fn: (r) => r["_field"] == "from_powerwall" or r["_field"] == "solar" or r["_field"] == "to_powerwall" or r["_field"] == "to_grid")
  |> sum()
  |> drop(columns: ["_start", "_stop"])
  |> pivot(rowKey: ["_measurement"], columnKey: ["_field"], valueColumn: "_value")
  |> map(fn: (r) => ({r with p_h: r.from_powerwall * ${kWhCost}}))
  |> map(fn: (r) => ({r with s_h: (r.solar - (r.to_powerwall + r.to_grid)) * ${kWhCost}}))
  |> map(fn: (r) => ({r with s_g: (r.to_grid) * ${kWhCost}}))
  |> map(fn: (r) => ({r with total: (r.p_h + r.s_h + r.s_g)}))
  |> map(fn: (r) => ({r with "blank": (" ")}))
  |> drop(columns: ["from_powerwall", "to_powerwall", "to_grid", "solar"])

```

**Reply by hulkster** [reply]
That works GREAT!!!

For solar_only, modify the lines in the "Saving Panels", so instead of "hard-wired" cents, you have something like this
_SELECT s - (tp + tg) FROM (SELECT sum("solar") * ${kWhCost} AS s,  sum("to_pw")  * ${kWhCost} AS tp, sum("to_grid") * ${kWhCost} AS tg FROM kwh.http WHERE  $timeFilter tz('America/Denver'))_

Thanks @ThePnuts
